pwiz_tools/Skyline/FileUI/ImportTransitionListColumnSelectDlg.cs

```diff
@@ -456,17 +730,171 @@ private void CheckEssentialColumn(Tuple<int, string> column)
             }
         }
 
+        private void CheckMoleculeColumns()
+        {
+            var columns = Importer.RowReader.Indices;
+            if (columns.PrecursorAdductColumn == -1 && columns.PrecursorChargeColumn == -1)
+            {
+                MissingEssentialColumns.Add(Resources.ImportTransitionListColumnSelectDlg_CheckMoleculeColumns_Precursor_Adduct_and_or_Precursor_Charge);
+            }
+
+            if (columns.MolecularFormulaColumn == -1 && columns.PrecursorColumn == -1)
+            {
+                MissingEssentialColumns.Add(Resources.ImportTransitionListColumnSelectDlg_CheckMoleculeColumns_Molecular_Formula_and_or_Precursor_m_z);
+            }
+        }
+
         public class DocumentChecked
         {
             public SrmDocument Document;
             public IdentityPath SelectPath;
             public List<MeasuredRetentionTime> IrtPeptides;
             public List<SpectrumMzInfo> LibrarySpectra;
             public List<PeptideGroupDocNode> PeptideGroups;
+            public List<string> ColumnHeaderList;
+            public bool IsSmallMoleculeList;
         }
 
         public DocumentChecked InsertionParams { get; private set; }
 
+        /// <summary>
```

Comment:
This kinda feels like it should be using that table that maps columns to SRMDocumentType?